source/common/http/conn_pool_base.h

```diff
@@ -69,6 +70,10 @@ class HttpConnPoolImplBase : public Envoy::ConnectionPool::ConnPoolImplBase,
                    Envoy::ConnectionPool::AttachContext& context) override;
 
   virtual CodecClientPtr createCodecClient(Upstream::Host::CreateConnectionData& data) PURE;
+
+protected:
+  Random::RandomGenerator& random_generator_;
+  Http::Protocol protocol_;
```

Comment:
const?

Reply:
it won't be for the ALPN subclass. :-)  OK if I leave it for now?